source/_integrations/conversation.markdown

```diff
@@ -24,73 +24,99 @@ The conversation integration allows you to converse with Home Assistant. You can
+      - sentences:
+          - "What is the humidity outside"
+```
 
+{% endraw %}
+
+To produce a response, create an `intent_script` entry in your `configuration.yaml` file:
+
+{% raw %}
+
+```yaml
+# Example configuration.yaml entry
 intent_script:
-  LivingRoomTemperature:
+  OutsideHumidity:
     speech:
-      text: It is currently {{ states.sensor.temperature }} degrees in the living room.
+      text: "It is currently {{ states("sensor.outside_humidity") }} percent humidity outside."
 ```
 
 {% endraw %}
 
 ## Adding advanced custom sentences
 
-Sentences can contain slots (marked with curly braces: `{name}`) and optional words (marked with square brackets: `[the]`). The values of slots will be passed on to the intent and are available inside the templates.
+Sentences can contain a specialized [template syntax](https://developers.home-assistant.io/docs/voice/intent-recognition/template-sentence-syntax), such as optional words and references to lists of names or numbers.
+
+The following configuration can handle sentences like "set living room volume to 100" and "change volume in the bedroom to 50".
 
-The following configuration can handle the following sentences:
+In `config/custom_sentences/en/media_player.yaml`:
 
-- Change the lights to red
-- Change the lights to green
-- Change the lights to blue
-- Change the lights to the color red
-- Change the lights to the color green
-- Change the lights to the color blue
+{% raw %}
+
+```yaml
+# Example media_player.yaml entry
+language: "en"
+intents:
+  SetVolume:
+    data:
+      - sentences:
+          - "(set|change) {media_player} volume to {volume} [percent]"
+          - "(set|change) volume in [the] {media_player} to {volume} [percent]"
+lists:
+  media_player:
+    values:
+      - in: "living room"
+        out: "media_player.living_room"
+      - in: "bedroom"
+        out: "media_player.bedroom"
+  volume:
+    range:
+      from: 0
+      to: 100
+```
+
+{% endraw %}
+
+The `media_player` list is referenced as `{media_player}` in the sentence templates. Its values contain both an `in` and `out` part, where `in` is the input text and `out` is the output value (`entity_id`) used in `intent_script` to call the `media_player.volume_set` service.
+
+In `configuration.yaml`:
 
 {% raw %}
 
 ```yaml
 # Example configuration.yaml entry
-conversation:
-  intents:
-    ColorLight:
-     - Change the lights to [the color] {color}
 intent_script:
-  ColorLight:
-    speech:
-      text: Changed the lights to {{ color }}.
+  SetVolume:
     action:
-      service: light.turn_on
+      service: "media_player.volume_set"
       data:
-        rgb_color:
-          - "{% if color == 'red' %}255{% else %}0{% endif %}"
-          - "{% if color == 'green' %}255{% else %}0{% endif %}"
-          - "{% if color == 'blue' %}255{% else %}0{% endif %}"
+        entity_id: "{{ media_player }}"
+        volume_level: "{{ volume / 100.0 }}"
+    speech:
+      text: "Volume changed to {{ volume }}"
```

Comment:
To make the text not contain a fraction

```suggestion
      text: "Volume changed to {{ volume | int }}"
```

Reply:
`volume` is already an `int` coming from the intent. Is this just in case it ends up being a `float`?